src/lib.rs

```diff
@@ -113,6 +116,7 @@ pub struct AbiFunction {
 
 /// Information about a single named function parameter.
 #[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+#[serde(deny_unknown_fields)]
```

Comment:
Seems not to work well here also because of the `#[serde(flatten)]`'d `typ`. But no worries, `AbiType`'s deny catches it.

Reply:
Yeah indeed, have just ran into deserialization issue as well